orchagent/portsorch.cpp

```diff
@@ -685,23 +694,50 @@ void PortsOrch::doPortTask(Consumer &consumer)
 
                 for (auto it = m_lanesAliasSpeedMap.begin(); it != m_lanesAliasSpeedMap.end();)
                 {
+                    bool port_created = false;
+
                     if (m_portListLaneMap.find(it->first) == m_portListLaneMap.end())
                     {
-                        if (!addPort(it->first, get<1>(it->second)))
+                        // work around to avoid syncd termination on SAI error due missing create_port SAI API
+                        // can be removed when SAI redis return NotImplemented error
+                        char *platform = getenv("platform");
+                        if (platform && strstr(platform, MLNX_PLATFORM_SUBSTRING))
                         {
-                            throw runtime_error("PortsOrch initialization failure.");
+                            if (!addPort(it->first, get<1>(it->second)))
+                            {
+                                throw runtime_error("PortsOrch initialization failure.");
+                            }
+
+                            port_created = true;
+                        }
+                        else
+                        {
+                            SWSS_LOG_NOTICE("Failed to create Port %s due to missing SAI create_port API.", get<0>(it->second).c_str());
                         }
                     }
+                    else
+                    {
+                        port_created = true;
+                    }
 
-                    if (!initPort(get<0>(it->second), it->first))
+                    if (port_created)
                     {
-                        throw runtime_error("PortsOrch initialization failure.");
+                        if (!initPort(get<0>(it->second), it->first))
+                        {
+                            throw runtime_error("PortsOrch initialization failure.");
+                        }
                     }
 
                     it = m_lanesAliasSpeedMap.erase(it);
                 }
             }
 
+            if (!m_portConfigDone)
+            {
+                it = consumer.m_toSync.erase(it);
+                continue;
+            }
```

Comment:
Not sure about this change.   So all port speed/admin_status/mtu set request for any specific port will be dropped before m_portConfigDone is set?

Reply:
this solves another bug introduced by previous PR. we do not process mtu, admin_status, speed before m_portConfigDone is set to true. It only make sense after initPort() is called.

Reply:
Please add a comment to that effect.

Reply:
There might be issue here,  currently port speed/admin_status/mtu "SET" operation in orchagent is triggered by Linux netlink message. If the request is dropped with "consumer.m_toSync.erase(it)",   it may never get set again?    There are multiple ports,  some ports may get netlink message before other ports get created in some extreme scenarios.

Reply:
it won't. the linux netlink messages will only come apter m_portConfigDone.